firmware/src/hal/sgx/test/nvmem/Makefile

```diff
@@ -0,0 +1,48 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2021 RSK Labs Ltd
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+include ../common/common.mk
+
+PROG = test.out
+OBJS = nvmem.o test_nvmem.o mock_secret_store.o
+
+all: $(PROG)
+
+$(PROG): $(OBJS)
+	$(CC) $(COVFLAGS) -o $@ $^
+
+test_nvmem.o: test_nvmem.c
+	$(CC) $(CFLAGS) -c -o $@ $^
+
+nvmem.o: $(SRCDIR)/nvmem.c
+	$(CC) $(CFLAGS) $(COVFLAGS) -c -o $@ $^
+
+mock_secret_store.o: $(MOCKDIR)/mock_secret_store.c
+	$(CC) $(CFLAGS) -c -o $@ $^
```

Comment:
We shouldn't need these targets. Check out other unit tests' makefiles to see what I mean.

Reply:
Sounds good. I took this pattern from ledger UI and signer tests, will double check others, thanks!

Reply:
targets removed, added `VPATH` to `common.mk` instead